src/qibolab/sequence.py

```diff
@@ -73,14 +95,35 @@ def concatenate(self, other: "PulseSequence") -> None:
             - necessary delays to synchronize channels
             - ``other``
         """
-        durations = {ch: self.channel_duration(ch) for ch in other.channels}
-        max_duration = max(durations.values(), default=0.0)
-        for ch, duration in durations.items():
-            delay = max_duration - duration
-            if delay > 0:
-                self.append((ch, Delay(duration=delay)))
+        _synchronize(self, other.channels)
         self.extend(other)
 
+    def align(self, channels: list[ChannelId]) -> Align:
+        """Introduce align commands to the sequence."""
+        align = Align()
+        for channel in channels:
+            self.append((channel, align))
+        return align
```

Comment:
> I have the feeling that, eventually, this is instruction should be part of the program, and always materialize in sequences as a delay (if the program gets compiled to a sequence). In which case, it would be more appropriate to have an `Align(channels=[...])`.

That was my original idea, because this is even closer to QUA, however I was not sure how to implement it since our `_PulseLike` objects do not have channels and every entry in the `PulseSequence` can only be associated with a single channel. That's why I came up with the current proposal, that satisfies these constraints. Maybe in 0.3 we can move towards that direction.